examples/channels-for-correlation/channels_correlation.bal

```diff
@@ -0,0 +1,40 @@
+import ballerina/http;
+
+//Defines a channel with json constrained type.
+channel<json> jsonChannel;
+
+service<http:Service> channelService bind { port: 9090 } {
+
+receive (endpoint caller, http:Request request) {
+       http:Response response = new;
+
+       //A key can be associated with a channel action, this is used to correlate receivers and senders.
+       string key = "123";
+
+       json result;
+       //Receive a message from the channel with given key.
+       //Execution waits here if the message is not available.
+       result <- jsonChannel, key;
+       //Received message is sent as the response.
+       response.setJsonPayload(result, contentType = "application/json");
+       _ = caller -> respond(response);
```

Comment:
Shall we handle the possible error here like in https://github.com/ballerina-platform/ballerina-examples/blob/master/examples/hello-world-service/hello_world_service.bal#L14?

Also shall we remove the space around `->` since it is a caller action? The standard agreed was to not have spaces around `->` if it is a caller action, but have spaces in other scenarios like worker interactions.